cob_twist_controller/src/inverse_jacobian_calculations/inverse_jacobian_calculation.cpp

```diff
@@ -63,3 +63,40 @@ Eigen::MatrixXd PInvBySVD::calculate(const AugmentedSolverParams& params,
     return pseudoInverseJacobian;
 }
 
+
+
+
+Eigen::MatrixXd PInvByLID::calculate(const AugmentedSolverParams& params,
+                                                 boost::shared_ptr<DampingBase> db,
+                                                 const Eigen::MatrixXd& jacobian) const
+{
+    Eigen::JacobiSVD<Eigen::MatrixXd> svd(jacobian, Eigen::ComputeThinU | Eigen::ComputeThinV);
+    double eps = params.eps;
+    double lambda;// = db->get_damping_factor();
+    double lambda_max = 0.5;
+    double beta = params.beta;
+    Eigen::VectorXd singularValues = svd.singularValues();
+    Eigen::VectorXd singularValuesInv = Eigen::VectorXd::Zero(singularValues.rows());
+    Eigen::MatrixXd pseudoInverseJacobian_1 = Eigen::MatrixXd::Zero(svd.matrixV().rows(),svd.matrixU().rows());
+
+    // Formula 15 Singularity-robust Task-priority Redundandancy Resolution
+    if((double)singularValues(singularValues.rows()-1) < eps){
+        lambda = sqrt( (1-pow((double)singularValues(singularValues.rows()-1)/eps,2)) * pow(lambda_max,2) );
+    }else{
+        lambda=0;
+    }
```

Comment:
if you implement this block within a new class derived from DampingBase, you could retrieve the lambda via ```db->get_damping_factor();``` and the inverse_jacobian class would be the same as all the others....

The DampingBase might need to be extended in that you can pass over a "direction_matrix" which is identity in default case and u_m*u_m^T in case of numerical filtering...